src/compiler/program.rs

```diff
@@ -235,7 +235,7 @@ fn find_source_file_worker(
     let file = helper_context.host.get_source_file(file_name);
 
     file.map(|file| {
-        let file: Rc<Node> = Rc::new(Rc::new(file).into());
+        let file: Rc<Node> = file.into();
```

Comment:
I couldn't implement `impl From<Rc<SourceFile>> for Rc<Node>` because I don't own the `Rc` type (so I was violating the "orphan rule"), so I adjusted this to be a `From<SourceFile> for Rc<Node>` instead (which knows that `SourceFile` is a special case where it gets its own direct `Rc<SourceFile>` wrapper before being wrapped in an `Rc<Node>`)